### PR TITLE
fetch_configlet: retry if curl request fails

### PR DESCRIPTION
**Pull request by glennj**
https://exercism-team.slack.com/archives/CAQP7JL3T/p1582614952007300

---

Reviewer Resources:

[Track Policies](https://github.com/exercism/bash/blob/master/POLICIES.md)

### CODE REVIEW

File: bin/fetch_configlet [hunk 1]
```diff
@@ -33,11 +33,31 @@ esac
 VERSION="$(get_version)"
 URL="${RELEASES}/download/${VERSION}/configlet-${OS}-${ARCH}.${EXT}"
 
+fetch_archive() {
+    local file_info=$1
+    for i in {0..4}; do
+        sleep $i
```
Comment (kotp) [inline bin/fetch_configlet]:
```suggestion
    for delay in {0..4}; do
        sleep $delay
```

Comment (IsaacG) [reply bin/fetch_configlet]: ...with variable expansions quoted, of course.

File: bin/fetch_configlet [hunk 2]
```diff
@@ -33,11 +33,31 @@ esac
...
+        curl -s --location "$URL" -o "$localfile"
+        if file "$localfile" 2>&1 | grep -q "$file_info"; then
+            # fetched successfully
```
Comment (IsaacG) [inline bin/fetch_configlet]: Shouldn't this just rely on curl's exit status?

Comment (glennj) [reply bin/fetch_configlet]: `errexit` is on, so the script will die on non-zero. 
Anyway the problem is that a small text file is fetched, not the (g)zip content we're expecting.

File: bin/fetch_configlet (hunk 1 above)
Comment (glennj) [reply bin/fetch_configlet]: It's an integer, and IFS has not been altered.

**Comment by glennj** [comment]
> Sight reviewed, and approvable, but not tested.

It was (kind of) field tested: https://travis-ci.org/exercism/bash/builds/655446128?utm_source=github_status&utm_medium=notification

**Comment by glennj** [comment]
@kotp @IsaacG @guygastineau I found the fix on another track: while querying the latest version, the `curl --head` results returned the header field names in lower case. I'll keep the retrying code for good measure, but the last commit updates the awk command for case insensitivity.

**Comment by glennj** [comment]
@IsaacG I'm not excluding you, but the UI won't let me re-request your review.

**Pull request merged**